xbmc/video/VideoInfoScanner.cpp

```diff
@@ -237,201 +246,159 @@ namespace VIDEO
-                                 DIR_FLAG_DEFAULTS);
-        // do not consider inner folders with .nomedia
-        items.erase(std::remove_if(items.begin(), items.end(),
-                                   [this](const CFileItemPtr& item) {
-                                     return item->m_bIsFolder && HasNoMedia(item->GetPath());
-                                   }),
-                    items.end());
-        items.Stack();
-
-        // check whether to re-use previously computed fast hash
-        if (!CanFastHash(items, regexps) || fastHash.empty())
-          GetPathHash(items, hash);
-        else
-          hash = fastHash;
-      }
-
-      if (StringUtils::EqualsNoCase(hash, dbHash))
-      { // hash matches - skipping
-        CLog::Log(LOGDEBUG, "VideoInfoScanner: Skipping dir '{}' due to no change{}",
-                  CURL::GetRedacted(strDirectory), !fastHash.empty() ? " (fasthash)" : "");
-        bSkip = true;
-      }
-      else if (hash.empty())
-      { // directory empty or non-existent - add to clean list and skip
-        CLog::Log(LOGDEBUG,
-                  "VideoInfoScanner: Skipping dir '{}' as it's empty or doesn't exist - adding to "
-                  "clean list",
-                  CURL::GetRedacted(strDirectory));
-        if (m_bClean)
-          m_pathsToClean.insert(m_database.GetPathId(strDirectory));
-        bSkip = true;
-      }
-      else if (dbHash.empty())
-      { // new folder - scan
-        CLog::Log(LOGDEBUG, "VideoInfoScanner: Scanning dir '{}' as not in the database",
-                  CURL::GetRedacted(strDirectory));
-      }
-      else
-      { // hash changed - rescan
-        CLog::Log(LOGDEBUG, "VideoInfoScanner: Rescanning dir '{}' due to change ({} != {})",
-                  CURL::GetRedacted(strDirectory), dbHash, hash);
-      }
-    }
-    else if (content == CONTENT_TVSHOWS)
-    {
-      if (m_handle)
-        m_handle->SetTitle(StringUtils::Format(g_localizeStrings.Get(20319), info->Name()));
-
-      if (foundDirectly && !settings.parent_name_root)
-      {
-        CDirectory::GetDirectory(strDirectory, items, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
-                                 DIR_FLAG_DEFAULTS);
-        items.SetPath(strDirectory);
-        GetPathHash(items, hash);
-        bSkip = true;
-        if (!m_database.GetPathHash(strDirectory, dbHash) || !StringUtils::EqualsNoCase(dbHash, hash))
-          bSkip = false;
-        else
-          items.Clear();
-      }
-      else
-      {
-        CFileItemPtr item(new CFileItem(URIUtils::GetFileName(strDirectory)));
-        item->SetPath(strDirectory);
-        item->m_bIsFolder = true;
-        items.Add(item);
-        items.SetPath(URIUtils::GetParentPath(item->GetPath()));
-      }
-    }
-    bool foundSomething = false;
-    if (!bSkip)
+  bool CVideoInfoScanner::DoScan(const std::string& strDirectory){
+    return false;
+  }
```

Comment:
Can you explain this change please?

Reply:
Yes well here is a problem indeed.
DoScan is a pure virtual function inherited from class CInfoScanner.

The problem is that the new scan logic is not doable with a simple "strDirectory" parameter as entry.
So I did not find a way to keep any purpose for this function but I cannot remove it without changing a lot of things such as not being a sub class of CInfoScanner.

This "DoScan" function is never called from outside so keeping this "dead" version is not great, but should not do harm.

Let me know if you see a better way.